Foundation/NSJSONSerialization.swift

```diff
@@ -716,21 +716,16 @@ private struct JSONReader {
                 defer { intEndPointer.deallocate(capacity: 1) }
                 let doubleEndPointer = UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>.allocate(capacity: 1)
                 defer { doubleEndPointer.deallocate(capacity: 1) }
-                
                 let intResult = strtol(startPointer, intEndPointer, 10)
                 let intDistance = startPointer.distance(to: intEndPointer[0]!)
                 let doubleResult = strtod(startPointer, doubleEndPointer)
                 let doubleDistance = startPointer.distance(to: doubleEndPointer[0]!)
-                
+                let roundedDouble = doubleResult.rounded()
```

Comment:
At minimum, this is best moved after the `guard doubleDistance > 0` statement that should be restored below. As mentioned earlier, you may not need to have this line at all.